File: app/src/main/java/com/kota/ASFramework/PageController/ASNavigationControllerView.java
package com.kota.ASFramework.PageController;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.view.View;
import android.widget.FrameLayout;
import com.kota.ASFramework.Model.ASSize;

/* loaded from: classes.dex */
public class ASNavigationControllerView extends ASPageView implements ASGestureViewDelegate {
  private ASPageView _background_view;
  private ASSize _content_size;
  private ASPageView _content_view;
  private ASGestureView _gesture_view;
  private ASNavigationController _page_controller;

  public ASNavigationControllerView(Context context) {
    super(context);
    this._background_view = null;
    this._content_view = null;
    this._gesture_view = null;
    this._content_size = new ASSize(0, 0);
    this._page_controller = null;
    initial(context);
  }

  public ASNavigationControllerView(Context context, AttributeSet attrs) {
    super(context, attrs);
    this._background_view = null;
    this._content_view = null;
    this._gesture_view = null;
    this._content_size = new ASSize(0, 0);
    this._page_controller = null;
    initial(context);
  }

  public ASNavigationControllerView(Context context, AttributeSet attrs, int defStyle) {
    super(context, attrs, defStyle);
    this._background_view = null;
    this._content_view = null;
    this._gesture_view = null;
    this._content_size = new ASSize(0, 0);
    this._page_controller = null;
    initial(context);
  }

  private void initial(Context context) {
    this._background_view = new ASPageView(context);
    this._background_view.setLayoutParams(new FrameLayout.LayoutParams(-1, -1));
    addView(this._background_view);
    this._content_view = new ASPageView(context);
    this._content_view.setLayoutParams(new FrameLayout.LayoutParams(-1, -1));
    addView(this._content_view);
    this._gesture_view = new ASGestureView(context);
    this._gesture_view.setLayoutParams(new FrameLayout.LayoutParams(-1, -1));
    this._gesture_view.setDelegate(this);
    addView(this._gesture_view);
  }

  public ASPageView getBackgroundView() {
    return this._background_view;
  }

  public ASPageView getContentView() {
    return this._content_view;
  }

  public ASSize getContentSize() {
    return this._content_size;
  }

  public void setPageController(ASNavigationController aController) {
    this._page_controller = aController;
  }

  @Override // android.view.View
  protected void onSizeChanged(int w, int h, int oldw, int oldh) {
    super.onSizeChanged(w, h, oldw, oldh);
    if (this._page_controller != null) {
      this._page_controller.onSizeChanged(w, h, oldw, oldh);
    }
  }

  @Override // com.kota.ASFramework.PageController.ASGestureViewDelegate
  public void onASGestureDispathTouchEvent(MotionEvent event) {
    this._content_view.dispatchTouchEvent(event);
  }

  @Override // com.kota.ASFramework.PageController.ASGestureViewDelegate
  public boolean onASGestureReceivedGestureUp() {
    if (this._page_controller != null) {
      return this._page_controller.onReceivedGestureUp();
    }
    return false;
  }

  @Override // com.kota.ASFramework.PageController.ASGestureViewDelegate
  public boolean onASGestureReceivedGestureDown() {
    if (this._page_controller != null) {
      return this._page_controller.onReceivedGestureDown();
    }
    return false;
  }

  @Override // com.kota.ASFramework.PageController.ASGestureViewDelegate
  public boolean onASGestureReceivedGestureLeft() {
    if (this._page_controller != null) {
      return this._page_controller.onReceivedGestureLeft();
    }
    return false;
  }

  @Override // com.kota.ASFramework.PageController.ASGestureViewDelegate
  public boolean onASGestureReceivedGestureRight() {
    if (this._page_controller != null) {
      return this._page_controller.onReceivedGestureRight();
    }
    return false;
  }
}

File: app/src/main/java/com/kota/asFramework/pageController/ASNavigationControllerView.kt
package com.kota.asFramework.pageController

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import com.kota.asFramework.model.ASSize

class ASNavigationControllerView : ASPageView, ASGestureViewDelegate {
    var backgroundView: ASPageView?
        private set
    val contentSize: ASSize?
    var contentView: ASPageView?
        private set
    private var gestureView: ASGestureView?
    private var pageController: ASNavigationController?

    constructor(context: Context) : super(context) {
        this.backgroundView = null
        this.contentView = null
        this.gestureView = null
        this.contentSize = ASSize(0, 0)
        this.pageController = null
        initial(context)
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        this.backgroundView = null
        this.contentView = null
        this.gestureView = null
        this.contentSize = ASSize(0, 0)
        this.pageController = null
        initial(context)
    }

    constructor(context: Context, attrs: AttributeSet?, defStyle: Int) : super(
        context,
        attrs,
        defStyle
    ) {
        this.backgroundView = null
        this.contentView = null
        this.gestureView = null
        this.contentSize = ASSize(0, 0)
        this.pageController = null
        initial(context)
    }

    private fun initial(context: Context) {
        this.backgroundView = ASPageView(context)
        this.backgroundView?.layoutParams = LayoutParams(-1, -1)
        addView(this.backgroundView)
        this.contentView = ASPageView(context)
        this.contentView?.layoutParams = LayoutParams(-1, -1)
        addView(this.contentView)
        this.gestureView = ASGestureView(context)
        this.gestureView?.layoutParams = LayoutParams(-1, -1)
        this.gestureView?.setDelegate(this)
        addView(this.gestureView)
    }

    fun setPageController(aController: ASNavigationController?) {
        this.pageController = aController
    }

    // android.view.View
    override fun onSizeChanged(w: Int, h: Int, oldw: Int, oldh: Int) {
        super.onSizeChanged(w, h, oldw, oldh)
        if (this.pageController != null) {
            this.pageController?.onSizeChanged(w, h, oldw, oldh)
        }
    }

    // com.kota.ASFramework.PageController.ASGestureViewDelegate
    override fun onASGestureDisPathTouchEvent(paramMotionEvent: MotionEvent?) {
        this.contentView?.dispatchTouchEvent(paramMotionEvent)
    }

    // com.kota.ASFramework.PageController.ASGestureViewDelegate
    override fun onASGestureReceivedGestureUp(): Boolean {
        if (this.pageController != null) {
            return this.pageController?.onReceivedGestureUp()
        }
        return false
    }

    // com.kota.ASFramework.PageController.ASGestureViewDelegate
    override fun onASGestureReceivedGestureDown(): Boolean {
        if (this.pageController != null) {
            return this.pageController?.onReceivedGestureDown()
        }
        return false
    }

    // com.kota.ASFramework.PageController.ASGestureViewDelegate
    override fun onASGestureReceivedGestureLeft(): Boolean {
        if (this.pageController != null) {
            return this.pageController?.onReceivedGestureLeft()
        }
        return false
    }

    // com.kota.ASFramework.PageController.ASGestureViewDelegate
    override fun onASGestureReceivedGestureRight(): Boolean {
        if (this.pageController != null) {
            return this.pageController?.onReceivedGestureRight()
        }
        return false
    }
}
